ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/authorization/services/TransactionServiceAuthorization.scala

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.server.api.authorization.services
+
+import com.daml.ledger.participant.state.v1.AuthService
+import com.digitalasset.grpc.adapter.utils.DirectExecutionContext
+import com.digitalasset.ledger.api.v1.transaction_service
+import com.digitalasset.ledger.api.v1.transaction_service.{
+  GetFlatTransactionResponse,
+  GetLedgerEndRequest,
+  GetLedgerEndResponse,
+  GetTransactionResponse,
+  GetTransactionTreesResponse,
+  GetTransactionsResponse,
+  TransactionServiceGrpc
+}
+import com.digitalasset.ledger.api.v1.transaction_service.TransactionServiceGrpc.TransactionService
+import com.digitalasset.platform.api.grpc.GrpcApiService
+import com.digitalasset.platform.server.api.ProxyCloseable
+import com.digitalasset.platform.server.api.authorization.ApiServiceAuthorization
+import io.grpc.ServerServiceDefinition
+import io.grpc.stub.StreamObserver
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.concurrent.Future
+
+class TransactionServiceAuthorization(
+    protected val service: TransactionService with AutoCloseable,
+    protected val authService: AuthService)
+    extends TransactionService
+    with ProxyCloseable
+    with GrpcApiService {
+
+  protected val logger: Logger = LoggerFactory.getLogger(TransactionService.getClass)
+
+  override def getTransactions(
+      request: transaction_service.GetTransactionsRequest,
+      responseObserver: StreamObserver[GetTransactionsResponse]): Unit =
+    // TODO(RC): To implement claims expiration, get the expiration date here and create a
```

Comment:
Simple curiosity: is `C` the initial of your middle name?

Reply:
Force of habit, my initials used to be `RC` 😄